api.go

```diff
@@ -0,0 +1,1997 @@
+	s.GrantRead = &v
+	return s
+}
+
+// SetGrantReadACP sets the GrantReadACP field's value.
+func (s *CreateBucketInput) SetGrantReadACP(v string) *CreateBucketInput {
+	s.GrantReadACP = &v
+	return s
+}
+
+// SetGrantWrite sets the GrantWrite field's value.
+func (s *CreateBucketInput) SetGrantWrite(v string) *CreateBucketInput {
+	s.GrantWrite = &v
+	return s
+}
+
+// SetGrantWriteACP sets the GrantWriteACP field's value.
+func (s *CreateBucketInput) SetGrantWriteACP(v string) *CreateBucketInput {
+	s.GrantWriteACP = &v
+	return s
+}
+
+// SetIsStaleAllowed sets the IsStaleAllowed field's value.
+func (s *CreateBucketInput) SetIsStaleAllowed(v bool) *CreateBucketInput {
+	s.IsStaleAllowed = &v
+	return s
+}
+
+// SetMetadataSearch sets the MetadataSearch field's value.
+func (s *CreateBucketInput) SetMetadataSearch(v string) *CreateBucketInput {
+	s.MetadataSearch = &v
+	return s
+}
+
+// SetNameSpace sets the NameSpace field's value.
+func (s *CreateBucketInput) SetNameSpace(v string) *CreateBucketInput {
+	s.NameSpace = &v
+	return s
+}
+
+// SetRetentionPeriod sets the RetentionPeriod field's value.
+func (s *CreateBucketInput) SetRetentionPeriod(v int64) *CreateBucketInput {
+	s.RetentionPeriod = &v
+	return s
+}
+
+// SetSSEEnabled sets the SSEEnabled field's value.
+func (s *CreateBucketInput) SetSSEEnabled(v bool) *CreateBucketInput {
+	s.SSEEnabled = &v
+	return s
+}
+
+// SetVPool sets the VPool field's value.
+func (s *CreateBucketInput) SetVPool(v string) *CreateBucketInput {
+	s.VPool = &v
+	return s
+}
+
+type DeleteBucketMetadataSearchInput struct {
+	_ struct{} `type:"structure"`
+
+	// Bucket is a required field
+	Bucket *string `location:"uri" locationName:"Bucket" type:"string" required:"true"`
+}
+
+// String returns the string representation
+func (s DeleteBucketMetadataSearchInput) String() string {
+	return awsutil.Prettify(s)
+}
+
+// GoString returns the string representation
+func (s DeleteBucketMetadataSearchInput) GoString() string {
+	return s.String()
+}
+
+// Validate inspects the fields of the type to determine if they are valid.
+func (s *DeleteBucketMetadataSearchInput) Validate() error {
+	invalidParams := request.ErrInvalidParams{Context: "DeleteBucketMetadataSearchInput"}
+	if s.Bucket == nil {
+		invalidParams.Add(request.NewErrParamRequired("Bucket"))
+	}
+
+	if invalidParams.Len() > 0 {
+		return invalidParams
+	}
+	return nil
+}
+
+// SetBucket sets the Bucket field's value.
+func (s *DeleteBucketMetadataSearchInput) SetBucket(v string) *DeleteBucketMetadataSearchInput {
+	s.Bucket = &v
+	return s
+}
+
+type DeleteBucketMetadataSearchOutput struct {
+	_ struct{} `type:"structure"`
+}
+
+// String returns the string representation
+func (s DeleteBucketMetadataSearchOutput) String() string {
+	return awsutil.Prettify(s)
+}
+
+// GoString returns the string representation
+func (s DeleteBucketMetadataSearchOutput) GoString() string {
+	return s.String()
+}
+
+type EmcEcsExtIndexableKey struct {
```

Comment:
Is the Emc necessary? If so, we should use EmcEcs everywhere we have our own variant, for consistency. If not, we should use Ecs for all.